logic/src/commonMain/kotlin/com/wire/kalium/logic/data/call/CallRepository.kt

```diff
@@ -134,6 +135,7 @@ internal class CallDataSource(
     private val teamRepository: TeamRepository,
     private val mlsClientProvider: MLSClientProvider,
     private val joinSubconversationUseCase: JoinSubconversationUseCase,
+    private val leaveSubconversationUseCase: LeaveSubconversationUseCase,
```

Comment:
I was thinking here if maybe it would be possible to not have the repository calling a usecase and rather the other way around?

Just so we can continue the flows we already have, such as:
`UI -> ViewModel -> UseCase -> Repository -> DAO/API`

Reply:
It would not make sense to have the view model call this use case since it's an internal use case. Maybe it shouldn't be a use case? Join and leave a subconversation could be actions on a repository.

Reply:
Hmmmm, maybe then having it as actions on a repository does make more sense but it is not something causing a problem. all good :)

Reply:
Yes, I'll think about how this can re-factored. But prefer not do it in this PR.